gold-zip-input.html

```diff
@@ -115,6 +117,31 @@
         type: String,
         value: "Zip Code"
       }
+    },
+
+    observers: [
+      '_computeValue(value)'
+    ],
+
+    _computeValue: function(value) {
+      var start = this.$.input.selectionStart;
+      var previousCharADash = this.value.charAt(start - 1) == '-';
+
+      // Remove any already-applied formatting.
+      value = value.replace(/-/g, '');
+
+      // Add a dash after the 5th character
+      if (value.length > 5) {
+        value = value.substr(0,5) + '-' + value.substr(5)
+      }
+      this.updateValueAndPreserveCaret(value.trim());
```

Comment:
`this.value = foo` will reset the value, and the caret will automagically jump to the end of the input (which, sadly, makes sense). `updateValueAndPreserveCaret` does a "if you are the kind of input that has a selectionStart, then restore it after we've reset it" dance, which I still shed tears about.